core/src/main/java/org/sonarsource/sonarlint/core/analyzer/sensor/LtsApiSensorContext.java

```diff
@@ -236,23 +210,9 @@ public Measure saveMeasure(InputFile inputFile, Measure measure) {
   @Override
   public Resource getResource(final InputPath inputPath) {
     if (inputPath.isFile()) {
-      File f = new File(inputPath.absolutePath()) {
-
-        @Override
-        public String getEffectiveKey() {
-          return inputPath.key();
-        }
-
-        @Override
-        public String getPath() {
-          return inputPath.absolutePath();
-        }
-
-        @Override
-        public org.sonar.api.resources.Directory getParent() {
-          return Directory.create(PathUtils.sanitize(inputPath.path().getParent().toString()));
-        }
-      };
+      File f = File.create(inputPath.absolutePath());
```

Comment:
This what make oldest versions of the Java plugin incompatible. I will modify the constructor to be protected so that we can override the class again.